src/Config/Variant.php

```diff
@@ -42,26 +42,35 @@ final class Variant
      */
     public const FIT_COVER = 'cover';
 
+    public const AVAILABLE_FITS = [
+        self::FIT_CROP,
+        self::FIT_SCALE_DOWN,
+        self::FIT_PAD,
+        self::FIT_CONTAIN,
+        self::FIT_COVER,
+    ];
+
     /**
      * The name of the variant. Right now it will be the name of the filter set
      */
     public string $name;
 
-    public int $width;
+    public ?int $width;
 
-    public int $height;
+    public ?int $height;
 
-    public string $fit;
+    public ?string $fit;
 
     /**
      * The name of the generator to use for this variant, i.e. 'cloudflare'
      */
     public string $generator;
 
-    public function __construct(string $name, string $generator, int $width, int $height, string $fit)
+    public function __construct(string $name, string $generator, ?int $width, ?int $height, ?string $fit)
     {
-        Assert::greaterThan($width, 0);
-        Assert::greaterThan($height, 0);
+        Assert::stringNotEmpty($name);
```

Comment:
When we have this on `$name` maybe we should have `Assert::nullOrStringNotEmpty` on `$fit`?